src/containers/ArticleListing/index.js

```diff
@@ -23,41 +34,92 @@ class ArticleListing extends React.Component {
       : <ArticleCard article={article} layout={layout} {...this.props} key={article.slug} />);
   };
 
-  renderArticles(articles) {
-    const { featured } = this.props;
+  renderArticles(articles, totalPages) {
+    const {
+      featured,
+    } = this.props;
     return (
       featured ? this.renderFeaturedArticles(articles)
-        : articles.map((article, index) => this.renderArticle(index, article))
+        : this.renderArticleList(articles, totalPages)
     );
   }
 
+  renderNoItems = () => {
+    const { emptyMessage } = this.props;
+    return <Empty message={emptyMessage} />;
+  };
+
+  renderArticleList = (articles, totalPages) => {
+    const { pageStart, infiniteScroll } = this.props;
+    const { page } = this.state;
+    return (
+      !articles || articles.length === 0 ? (
+        this.renderNoItems()
+      )
+        : (
+          <InfiniteScroll
+            pageStart={pageStart}
+            initialLoad={false}
+            loadMore={p => this.loadMore(p, totalPages)}
```

Comment:
Add tests to cover this line

Reply:
Thanks for the review. Let me work on it.

Reply:
I have implemented the changes as requested.